rsocket-transport-ktor/src/commonMain/kotlin/io/rsocket/kotlin/transport/ktor/TcpConnection.kt

```diff
@@ -57,14 +57,20 @@ internal class TcpConnection(private val socket: Socket) : Connection, Coroutine
                 while (isActive) {
                     val length = readPacket(3).readLength()
                     val packet = readPacket(length)
-                    receiveChannel.send(packet)
+                    try {
```

Comment:
Any reason you don't use closeOnError here?

Reply:
it's internal in rsocket-core :)
not sure, if it's needed to be public
this will be simplified with coroutines 1.5.0 (with new channel operators)